observer/baseNodeProvider.go

```diff
@@ -148,80 +207,119 @@ func (bnp *baseNodeProvider) isReceivedSyncedNodeExistent(receivedNode *data.Nod
 	return false
 }
 
+func (bnp *baseNodeProvider) isReceivedSyncedNodeExistentAsFallback(receivedNode *data.NodeData) bool {
+	for _, node := range bnp.syncedFallbackNodes {
+		if node.Address == receivedNode.Address && node.ShardId == receivedNode.ShardId {
+			return true
+		}
+	}
+
+	return false
+}
+
 func (bnp *baseNodeProvider) addToOutOfSyncUnprotected(node *data.NodeData) {
-	if bnp.outOfSyncNodes == nil {
-		bnp.outOfSyncNodes = make([]*data.NodeData, 0)
+	source := &bnp.outOfSyncNodes
+	if node.IsFallback {
+		source = &bnp.outOfSyncFallbackNodes
 	}
 
-	for _, oosNode := range bnp.outOfSyncNodes {
+	if *source == nil {
+		*source = make([]*data.NodeData, 0)
+	}
+
+	for _, oosNode := range *source {
 		if oosNode.Address == node.Address && oosNode.ShardId == node.ShardId {
 			return
 		}
 	}
 
-	bnp.outOfSyncNodes = append(bnp.outOfSyncNodes, node)
+	*source = append(*source, node)
 }
 
 func (bnp *baseNodeProvider) removeOutOfSyncNodesUnprotected(
 	outOfSyncNodes []*data.NodeData,
 	syncedNodesMap map[uint32][]*data.NodeData,
+	syncedFallbackNodesMap map[uint32][]*data.NodeData,
 ) {
 	if len(outOfSyncNodes) == 0 {
 		bnp.outOfSyncNodes = make([]*data.NodeData, 0)
+		bnp.outOfSyncFallbackNodes = make([]*data.NodeData, 0)
 		return
 	}
 
 	for _, outOfSyncNode := range outOfSyncNodes {
-		if len(syncedNodesMap[outOfSyncNode.ShardId]) < 1 {
-			log.Warn("cannot remove observer as not enough will remain in shard",
+		hasOneSyncedNode := len(syncedNodesMap[outOfSyncNode.ShardId]) >= 1
+		hasEnoughSyncedFallbackNodes := len(syncedFallbackNodesMap[outOfSyncNode.ShardId]) > 1
```

Comment:
it is conditioned to be at least 2, because we check if the current node is fallback and if it is safe to be removed.. If there are at least 2 fallbacks, we are safe to remove the current one. Otherwise, we'll go further and we'll act accordingly.